src/main/java/kitchenpos/application/MenuService.java

```diff
@@ -1,84 +1,85 @@
-        this.productDao = productDao;
+        this.menuRepository = menuRepository;
+        this.menuGroupRepository = menuGroupRepository;
+        this.productRepository = productRepository;
     }
 
-    @Transactional
-    public Menu create(final Menu menu) {
-        final BigDecimal price = menu.getPrice();
-
-        if (Objects.isNull(price) || price.compareTo(BigDecimal.ZERO) < 0) {
-            throw new IllegalArgumentException();
-        }
+    public Long create(final MenuCreateRequest request) {
+        validateExistedMenuGroup(request);
+        final List<Long> productIds = extractIds(request.getMenuProducts());
+        final MenuGroup menuGroup = menuGroupRepository.getById(request.getMenuGroupId());
+        final List<Product> products = productRepository.findByIdIn(productIds);
+        validateAllProductsFound(productIds, products);
+        final Map<Product, Integer> productWithQuantity = makeProductWithQuantity(request.getMenuProducts(), products);
+        final Menu menu = Menu.of(request.getName(), request.getPrice(), menuGroup, productWithQuantity);
+        return menuRepository.save(menu).getId();
+    }
 
-        if (!menuGroupDao.existsById(menu.getMenuGroupId())) {
-            throw new IllegalArgumentException();
-        }
+    private static List<Long> extractIds(final List<MenuProductRequest> menuCreateRequest) {
+        return menuCreateRequest.stream()
+                .map(MenuProductRequest::getProductId)
+                .collect(Collectors.toList());
+    }
 
-        final List<MenuProduct> menuProducts = menu.getMenuProducts();
+    @Transactional(readOnly = true)
+    public List<MenuResponse> findAll() {
+        return menuRepository.findAll().stream()
+                .map(MenuResponse::from)
+                .collect(Collectors.toList());
+    }
 
-        BigDecimal sum = BigDecimal.ZERO;
-        for (final MenuProduct menuProduct : menuProducts) {
-            final Product product = productDao.findById(menuProduct.getProductId())
-                    .orElseThrow(IllegalArgumentException::new);
-            sum = sum.add(product.getPrice().multiply(BigDecimal.valueOf(menuProduct.getQuantity())));
-        }
+    private Map<Product, Integer> makeProductWithQuantity(
+            final List<MenuProductRequest> menuProductRequests,
+            final List<Product> products
+    ) {
+        final Map<Long, Integer> productIdToQuantity = menuProductRequests.stream()
+                .collect(Collectors.toMap(MenuProductRequest::getProductId, MenuProductRequest::getQuantity));
+
+        return products.stream()
+                .collect(Collectors.toMap(
+                        product -> product,
+                        product -> productIdToQuantity.get(product.getId())
+                ));
+    }
 
-        if (price.compareTo(sum) > 0) {
+    private static void validateAllProductsFound(final List<Long> productIdToQuantity, final List<Product> products) {
```

Comment:
```suggestion
    private static void validateAllProductsFound(final List<Long> productIds, final List<Product> products) {
```
실제 검증 로직을 보면 그냥 id를 담고 있는 리스트네요. quantity 붙이면 코드 읽는데 오히려 혼란만 가져옵니다.

Reply:
파라미터명도 꼼꼼히 확인하겠습니다!